pkg/broker/awsbroker.go

```diff
@@ -241,127 +210,142 @@ func (b *AwsBroker) ValidateBrokerAPIVersion(version string) error {
-					pvals := make(map[string]interface{})
-					for pk, pv := range param.(map[interface{}]interface{}) {
-						switch pk {
-						case "name":
-							name = pv.(string)
-						case "required":
-							required = pv.(bool)
-						case "type":
-							switch pv {
-							case "enum":
-								pvals[pk.(string)] = "string"
-							case "int":
-								pvals[pk.(string)] = "integer"
-							default:
-								pvals[pk.(string)] = pv
-							}
-						case "updatable":
-							updatable = pv.(bool)
-						default:
-							pvals[pk.(string)] = pv
-						}
-					}
-					propsForCreate[name] = pvals
-					if required {
-						requiredForCreate = append(requiredForCreate, name)
-					}
-					if updatable {
-						propsForUpdate[name] = pvals
-						if required {
-							requiredForUpdate = append(requiredForUpdate, name)
-						}
+			}
+			required := false
+			if paramValue.(map[string]interface{})["required"] != nil {
+				required = paramValue.(map[string]interface{})["required"].(bool)
+			}
+			if include {
+				openshiftFormCreate = openshiftFormAppend(openshiftFormCreate, paramName, paramValue.(map[string]interface{}))
+				createParam := map[string]interface{}{}
+				for nk, nv := range paramValue.(map[string]interface{}) {
+					createParam[nk] = nv
+				}
+				for planDefaultParam, planDefaultValue := range p.ParameterDefaults {
+					if planDefaultParam == paramName {
+						glog.V(10).Infof("Updating default with plan default for plan %q param %q\n", k, paramName)
+						createParam["default"] = planDefaultValue
 					}
 				}
-				plan.Schemas = &osb.Schemas{
-					ServiceInstance: &osb.ServiceInstanceSchema{
-						Create: &osb.InputParametersSchema{
-							Parameters: map[string]interface{}{
-								"type":       "object",
-								"properties": propsForCreate,
-								"$schema":    "http://json-schema.org/draft-06/schema#",
-							},
-						},
-					},
+				for _, v := range []string{"required", "display_group"} {
+					delete(createParam, v)
 				}
-				if len(requiredForCreate) > 0 {
-					// Cloud Foundry does not allow "required" to be an empty slice
-					plan.Schemas.ServiceInstance.Create.Parameters.(map[string]interface{})["required"] = requiredForCreate
+				propsForCreate[paramName] = createParam
+				if required {
+					requiredForCreate = append(requiredForCreate, paramName)
 				}
-				if len(propsForUpdate) > 0 {
-					plan.Schemas.ServiceInstance.Update = &osb.InputParametersSchema{
-						Parameters: map[string]interface{}{
-							"type":       "object",
-							"properties": propsForUpdate,
-							"$schema":    "http://json-schema.org/draft-06/schema#",
-						},
+				if stringInSlice(paramName, sd.Metadata.Spec.UpdatableParameters) {
+					openshiftFormUpdate = openshiftFormAppend(openshiftFormUpdate, paramName, paramValue.(map[string]interface{}))
+					updateParam := paramValue.(map[string]interface{})
+					for _, v := range []string{"required", "display_group", "default"} {
+						delete(updateParam, v)
```

Comment:
This bit's buggy, as it will delete the keys from the original `paramValue` map. Maybe something like this instead:
```go
					updateParam := make(map[string]interface{})
					for k, v := range paramValue.(map[string]interface{}) {
						if k != "required" && k != "display_group" && k != "default" {
							updateParam[k] = v
						}
					}
```

There may be a similar problem on line 261, but I'm not positive (I haven't tested the OpenShift stuff).

https://github.com/awslabs/aws-servicebroker/pull/48/files#diff-185f61f44fd42ec6610b55fc8219880cR261

Reply:
Good catch @vsomayaji. fixed as per your suggestion.  going to merge